src/scripts/sync_subclassof.py

```diff
@@ -0,0 +1,573 @@
+    that do not exist in the 2nd ontology, or (b) the edges should be in both ontologies, but if the inputss
+    to this pipeline are out of sync, the mapped terms did not appear in ns_map.
+    """
+    rels2: Set[RELATIONSHIP] = set()
+    failed_lookups = set()
+    for x in rels:
+        try:
+            # For Source -> Mondo, should only be 1 sub and 1 obj, subs and objs should be strings
+            # For Mondo -> Source, subs and objs should be lists of 1+ items
+            subs, pred, objs = ns_map[x[0]], x[1], ns_map[x[2]]
+            subs: List[CURIE] = [subs] if isinstance(subs, str) else subs
+            objs: List[CURIE] = [objs] if isinstance(objs, str) else objs
+            for s in subs:
+                for o in objs:
+                    rels2.add((s, pred, o))
+        except KeyError:
+            failed_lookups.add(x)
+
+    return rels2, failed_lookups if return_failed_lookups else rels2
+
+
+def sync_subclassof(
+    outpath_added: str = EX_DEFAULTS['outpath_added'], outpath_confirmed: str = EX_DEFAULTS['outpath_confirmed'],
+    outpath_added_obsolete: str = EX_DEFAULTS['outpath_added_obsolete'],
+    mondo_db_path: str = EX_DEFAULTS['mondo_db_path'], mondo_ingest_db_path: str = EX_DEFAULTS['mondo_ingest_db_path'],
+    mondo_mappings_path: str = EX_DEFAULTS['mondo_mappings_path'],
+    onto_config_path: str = EX_DEFAULTS['onto_config_path'],
+    outpath_direct_in_mondo_only: str = EX_DEFAULTS['outpath_in_mondo_only'], use_cache=False, verbose=True
+):
+    """Run"""
+    source_name = os.path.basename(outpath_added).split('.')[0]
+    owned_prefix_map: PREFIX_MAP = get_owned_prefix_map(onto_config_path)
+    mondo_db: SqlImplementation = get_adapter(mondo_db_path)
+    ingest_db: SqlImplementation = get_adapter(mondo_ingest_db_path)
+    ontology_name = list(owned_prefix_map.keys())[0]
+    print(f'Running: Synchronization pipeline - subClassOf - {ontology_name}')
+
+    # Source terms and mappings ----------------------------------------------------------------------------------------
+    # - ss_df: Dataframe for mondo.sssom.tsv
+    # - mondo_source_map: Map of Mondo IDs to 1+ source IDs each
+    # - source_mondo_map: Map of source IDs to 1 Mondo ID each
+    # - mondo_label_lookup: For looking up Mondo labels
+    # - source_data_map: keys=source ids, values(mondo_id (if any), source_id, mondo_label (if any))
+    # - mondo_data_map: keys=mondo ids, values=(mondo_id, source_id (if any), mondo_label)
+    # - mondo_ids: List of set of Mondo IDs from mondo.sssom.tsv
+    # - source_ids: List of set of source IDs from mondo.sssom.tsv
+    print('- Collecting all source terms and mondo mappings')
+    # - Mondo labels
+    mondo_ids_all: List[CURIE] = [
+        x for x in [y for y in mondo_db.entities(filter_obsoletes=False)] if x.startswith('MONDO')]
+    mondo_label_lookup: Dict[CURIE, str] = {x[0]: x[1] for x in [y for y in mondo_db.labels(mondo_ids_all)]}
```

Comment:
## No missing labels for Case 5
I changed the way I was doing label lookups to fetch from the ontology rather than the SSSOM.